feedsx/src/main/java/com/likeminds/feedsx/topic/viewmodel/LMFeedTopicSelectionViewModel.kt

```diff
@@ -0,0 +1,129 @@
+package com.likeminds.feedsx.topic.viewmodel
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.likeminds.feedsx.topic.model.LMFeedAllTopicsViewData
+import com.likeminds.feedsx.topic.model.LMFeedTopicViewData
+import com.likeminds.feedsx.utils.ViewDataConverter
+import com.likeminds.feedsx.utils.coroutine.launchIO
+import com.likeminds.feedsx.utils.model.BaseViewType
+import com.likeminds.likemindsfeed.LMFeedClient
+import com.likeminds.likemindsfeed.topic.model.GetTopicRequest
+import javax.inject.Inject
+
+class LMFeedTopicSelectionViewModel @Inject constructor() : ViewModel() {
+    companion object {
+        const val SEARCH_TYPE = "name"
+        const val PAGE_SIZE = 10
+    }
+
+    private val lmFeedClient = LMFeedClient.getInstance()
+
+    //first -> page
+    //second -> list of topics
+    private val _topicsViewData = MutableLiveData<Pair<Int, List<BaseViewType>>>()
+    val topicsViewData: LiveData<Pair<Int, List<BaseViewType>>> = _topicsViewData
+
+    private val _errorMessage = MutableLiveData<String?>()
+    val errorMessage: LiveData<String?> = _errorMessage
+
+    private val selectedTopics by lazy {
+        HashMap<String, LMFeedTopicViewData>()
+    }
+
+    private val disabledTopics by lazy {
+        ArrayList<LMFeedTopicViewData>()
+    }
+
+    fun setPreviousSelectedTopics(selectedTopics: List<LMFeedTopicViewData>?) {
+        if (selectedTopics.isNullOrEmpty()) return
+        selectedTopics.forEach { topic ->
+            this.selectedTopics[topic.id] = topic
+        }
+    }
+
+    fun setPreviousDisabledTopics(disabledTopics: List<LMFeedTopicViewData>?) {
+        if (disabledTopics.isNullOrEmpty()) return
+        val updatedDisabledTopics = disabledTopics.map {
+            it.toBuilder().isSelected(true).build()
+        }
+        this.disabledTopics.addAll(updatedDisabledTopics)
+    }
+
+    fun getTopics(
+        showAllTopicFilter: Boolean,
+        showEnabledTopicOnly: Boolean,
+        page: Int,
+        searchString: String? = null
+    ) {
+        viewModelScope.launchIO {
+
+            val requestBuilder = GetTopicRequest.Builder()
+                .page(page)
+                .pageSize(PAGE_SIZE)
+
+            if (!searchString.isNullOrEmpty()) {
+                requestBuilder.search(searchString)
+                    .searchType(SEARCH_TYPE)
+            }
+
+            if (showEnabledTopicOnly) {
+                requestBuilder.isEnabled(true)
+            }
```

Comment:
is this check required? why not send `showEnabledTopicOnly` as a variable

Reply:
Yes, it is required